### PR TITLE
use java 8 features: List.sort, Comparator.comparing and others

### PR DESCRIPTION
**Pull request by AlexElin**
use java 8 features: List.sort, Comparator.comparing and others

### CODE REVIEW

File: pitest-entry/src/main/java/org/pitest/bytecode/analysis/MethodTree.java [hunk 1]
```diff
@@ -74,9 +73,8 @@ public boolean isSynthetic() {
 
   private List<AbstractInsnNode> createInstructionList() {
     final List<AbstractInsnNode> list = new LinkedList<>();
-    final ListIterator<AbstractInsnNode> it = this.rawNode.instructions.iterator();
-    while (it.hasNext()) {
-        list.add(it.next());
+    for (AbstractInsnNode abstractInsnNode : this.rawNode.instructions) {
+      list.add(abstractInsnNode);
     }
     this.lazyInstructions = new ArrayList<>(list);
```
Comment (StefanPenndorf) [inline pitest-entry/src/main/java/org/pitest/bytecode/analysis/MethodTree.java:79]: I'm wondering why there is a double copy of the list. First when iterating over this.rawNode.instructions those instructions will be "copied"  into list. Afterwards list will be copied into new ArraList<>(list) . I don't think the second copy is necessary. We should use ArrayList from the beginning. Is the size/number of instructions known? Would make sense to initialize the ArrayList with the number of items before copying from instructions.

File: pitest-entry/src/main/java/org/pitest/coverage/CoverageData.java [hunk 2]
```diff
@@ -252,13 +251,7 @@ private TestInfo createTestInfo(final Description description,
   }
 
   private int getNumberOfCoveredLines(final ClassName clazz) {
```
Comment (StefanPenndorf) [inline pitest-entry/src/main/java/org/pitest/coverage/CoverageData.java:253]: The name of the method and the implementation do not fit, what do you think (is not your fault but as I review the code that came to my mind) the method returns theNumberOfCoveredTests(...) not the number of covered lines. Need to check usages of the method later.

File: pitest/src/test/java/org/pitest/functional/FCollectionTest.java [hunk 3]
```diff
@@ -60,9 +60,9 @@ public void shouldApplyForEachToAllItems() {
     final List<Integer> actual = new ArrayList<>();
     final Consumer<Integer> e = a -> actual.add(a);
 
-    FCollection.forEach(this.is, e);
+      this.is.forEach(e);
 
-    assertEquals(this.is, actual);
+      assertEquals(this.is, actual);
```
Comment (StefanPenndorf) [inline pitest/src/test/java/org/pitest/functional/FCollectionTest.java:65]: The additional indention is strange ... but doesn't harm anybody ;-) Maybe the whole method is not correctly indented? I didn't check.

Comment (AlexElin) [reply pitest/src/test/java/org/pitest/functional/FCollectionTest.java]: fixed

File: pitest-entry/src/main/java/org/pitest/bytecode/analysis/MethodTree.java (hunk 1 above)
Comment (AlexElin) [reply pitest-entry/src/main/java/org/pitest/bytecode/analysis/MethodTree.java]: removed double copy.

File: pitest-entry/src/main/java/org/pitest/coverage/CoverageData.java (hunk 2 above)
Comment (AlexElin) [reply pitest-entry/src/main/java/org/pitest/coverage/CoverageData.java]: I think `getTestsForClassName` has wrong name.
Tried to resolve this problem in latest commit.
Please, take a look

Comment (StefanPenndorf) [reply pitest-entry/src/main/java/org/pitest/coverage/CoverageData.java]: LGTM

**Pull request merged**